src/plugins/expressions/common/execution/types.ts

```diff
@@ -42,6 +42,11 @@ export interface ExecutionContext<Input = unknown, InspectorAdapters = DefaultIn
    */
   types: Record<string, ExpressionType>;
 
+  /**
+   * prevents caching in the current execution.
+   */
+  disableCache?: boolean;
```

Comment:
this should already be provided to all functions on the executionContext.

if some function would to do some internal caching (esaggs with make it slow) this would indicate that it should do it.